### ISSUE TITLE
[ERROR] IndexError: list index out of range

### ISSUE THREAD

**Issue by ocoot** [PokemonGoF/PokemonGo-Bot#3121]
Traceback (most recent call last):
  File "pokecli.py", line 503, in <module>
    main()
  File "pokecli.py", line 75, in main
    bot.start()
  File "/root/PokemonGo-Bot/pokemongo_bot/__init__.py", line 75, in start
    self._setup_api()
  File "/root/PokemonGo-Bot/pokemongo_bot/__init__.py", line 678, in _setup_api
    self.update_web_location()
  File "/root/PokemonGo-Bot/pokemongo_bot/__init__.py", line 492, in update_web_location
    gym_longitude=fort.get('longitude')
  File "/root/PokemonGo-Bot/src/pgoapi/pgoapi/pgoapi.py", line 122, in function
    return request.call()
  File "/root/PokemonGo-Bot/pokemongo_bot/api_wrapper.py", line 106, in call
    result = self._call()
  File "/root/PokemonGo-Bot/pokemongo_bot/api_wrapper.py", line 63, in _call
    return PGoApiRequest.call(self)
  File "/root/PokemonGo-Bot/src/pgoapi/pgoapi/pgoapi.py", line 220, in call
    response = request.request(self._api_endpoint, self._req_method_list, self.get_position())
  File "/root/PokemonGo-Bot/src/pgoapi/pgoapi/rpc_api.py", line 129, in request
    response_dict = self._parse_main_response(response, subrequests)
  File "/root/PokemonGo-Bot/src/pgoapi/pgoapi/rpc_api.py", line 359, in _parse_main_response
    response_proto_dict = self._parse_sub_responses(response_proto, subrequests, response_proto_dict)
  File "/root/PokemonGo-Bot/src/pgoapi/pgoapi/rpc_api.py", line 376, in _parse_sub_responses
    request_entry = subrequests_list[i]
IndexError: list index out of range

what this is? :/

**Comment by SpencerCarstens** [comment]
I just got the same thing. 

Works on iOS app still, so not a ban.

**Comment by tarpan2** [comment]
Getting this aswell. Bot runs for like 1 Minute, then exits with IndexError on that subrequests_list

**Comment by Kassadd** [comment]
Same, no changes in code or config

**Comment by z4q** [comment]
me also get 

2016-08-09 02:32:27,094 [PokemonGoBot] [INFO] Found encrypt.so! Platform: darwin Bot root directory: /Users/z4q/Code/pokemongo/bot01
2016-08-09 02:32:27,095 [PokemonGoBot] [INFO] 
2016-08-09 02:32:31,235 [PokemonGoBot] [INFO] [bot_start] Starting bot...
2016-08-09 02:32:31,998 [       cli] [INFO] 
2016-08-09 02:32:31,999 [       cli] [INFO] Ran for 0:00:08
2016-08-09 02:32:31,999 [       cli] [INFO] Total XP Earned: 0  Average: 0.00/h
2016-08-09 02:32:31,999 [       cli] [INFO] Travelled 0.00km
2016-08-09 02:32:31,999 [       cli] [INFO] Visited 0 stops
2016-08-09 02:32:31,999 [       cli] [INFO] Encountered 0 pokemon, 0 caught, 0 released, 0 evolved, 0 never seen before
2016-08-09 02:32:31,999 [       cli] [INFO] Threw 0 pokeballs
2016-08-09 02:32:31,999 [       cli] [INFO] Earned 0 Stardust
2016-08-09 02:32:31,999 [       cli] [INFO] 
2016-08-09 02:32:31,999 [       cli] [INFO] Highest CP Pokemon: 
2016-08-09 02:32:32,000 [       cli] [INFO] Most Perfect Pokemon: 
Traceback (most recent call last):
  File "pokecli.py", line 502, in <module>
    main()
  File "pokecli.py", line 133, in main
    raise e
IndexError: list index out of range

**Comment by Philipp59** [comment]
me2 error appears since 10min ago no changed were made

**Comment by GitPupper** [comment]
It works now, go try.

**Comment by omeryagmurlu** [comment]
Same here, just to note, I get the same error also on pokemongo-map-node, but pokemongo-bot, which is java, works fine.

**Comment by ocoot** [comment]
still error on me

**Comment by ocoot** [comment]
using latest dev version

**Comment by stefansaraev** [comment]
perhaps servers unstable. please have some patience and try later :)

**Comment by rosalf** [comment]
same here, last 10 minutes it crashed out, Traceback (most recent call last):
  File "pokecli.py", line 502, in <module>
    main()
  File "pokecli.py", line 133, in main
    raise e
IndexError: list index out of range
2016-08-08 20:42:12,175 [sentry.errors] [ERROR] Sentry responded with an error: 'ascii' codec can't decode byte 0x9c in position 1: ordinal not in range(128) (url: https://app.getsentry.com/api/90254/store/)
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/raven/transport/threaded.py", line 174, in send_sync
    super(ThreadedHTTPTransport, self).send(data, headers)
  File "/usr/local/lib/python2.7/site-packages/raven/transport/http.py", line 47, in send
    ca_certs=self.ca_certs,
  File "/usr/local/lib/python2.7/site-packages/raven/utils/http.py", line 66, in urlopen
    return opener.open(url, data, timeout)
  File "/Library/Python/2.7/site-packages/future/backports/urllib/request.py", line 494, in open
    response = self._open(req, data)
  File "/Library/Python/2.7/site-packages/future/backports/urllib/request.py", line 512, in _open
    '_open', req)
  File "/Library/Python/2.7/site-packages/future/backports/urllib/request.py", line 466, in _call_chain
    result = func(*args)
  File "/usr/local/lib/python2.7/site-packages/raven/utils/http.py", line 46, in https_open
    return self.do_open(ValidHTTPSConnection, req)
  File "/Library/Python/2.7/site-packages/future/backports/urllib/request.py", line 1284, in do_open
    h.request(req.get_method(), req.selector, req.data, headers)
  File "/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 1057, in request
    self._send_request(method, url, body, headers)
  File "/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 1097, in _send_request
    self.endheaders(body)
  File "/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 1053, in endheaders
    self._send_output(message_body)
  File "/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 895, in _send_output
    msg += message_body
UnicodeDecodeError: 'ascii' codec can't decode byte 0x9c in position 1: ordinal not in range(128)
2016-08-08 20:42:12,176 [sentry.errors.uncaught] [ERROR] [u'IndexError: list index out of range', u'  File "pokecli.py", line 502, in <module>', u'  File "pokecli.py", line 133, in main']

**Comment by Philipp59** [comment]
unknown 7 incoming : D

**Comment by mrthrowaway** [comment]
@ocoot could you change the title to something like 'IndexError: list index out of range'? Perhaps more of those affected can see this and share info

**Comment by AllYourBasesAreBelongToUs** [comment]
Same here, this is super weird. Here is what happens on my three bots A, B, C.

A: Ran fine throughout this episode without any problem:
B and C: Crashed *simultaneously* when receiving an abnormally short response from the server.

I then tried to restart B and C a couple times and they kept receiving too short responses. I setup the bot to keep trying after 30 sec when that happened, but always receiving these shorter responses.
Then I waited for 5-10 min, and now I can use B and C again.

So to me it's clear: Niantic is playing with our bots using the couple fields of the API that we still haven't figured out to see what accounts and how many are dropping.
The fact that A kept running is maybe because they changed the server response only on a limited set of accounts, so as to not disturb / flag their test to strongly.
But, hey I might be completely wrong.

**Comment by scarfield** [comment]
Mine ran for 2.5 hours before this showed up. Connection issues with the servers. If I run it again the error comes immediately. So it's probably (hopefully) something that will sort itself out.
```
2016-08-08 20:25:16,149 [  SpinFort] [INFO] [spun_pokestop] Spun pokestop Alien et Femme. Experience awarded: 50. Items awarded: {u'Pokeball': 3}
2016-08-08 20:25:19,187 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Majoor Bosshardtburgh - 0.06km (attraction of lure 0.14km)
2016-08-08 20:25:45,314 [pokemongo_bot.api_wrapper] [WARNING] Server seems to be busy or offline - try again - 4/15
2016-08-08 20:25:46,532 [pokemongo_bot.api_wrapper] [WARNING] Server seems to be busy or offline - try again - 5/15
2016-08-08 20:25:48,188 [pokemongo_bot.api_wrapper] [WARNING] Server seems to be busy or offline - try again - 6/15
2016-08-08 20:26:31,415 [       cli] [INFO]
2016-08-08 20:26:31,419 [       cli] [INFO] Ran for 2:25:53
2016-08-08 20:26:31,420 [       cli] [INFO] Total XP Earned: 10650  Average: 4380.44/h
2016-08-08 20:26:31,421 [       cli] [INFO] Travelled 1.66km
2016-08-08 20:26:31,422 [       cli] [INFO] Visited 41 stops
2016-08-08 20:26:31,422 [       cli] [INFO] Encountered 40 pokemon, 40 caught, 41 released, 0 evolved, 0 never seen before
2016-08-08 20:26:31,423 [       cli] [INFO] Threw 40 pokeballs
2016-08-08 20:26:31,424 [       cli] [INFO] Earned 4673 Stardust
2016-08-08 20:26:31,424 [       cli] [INFO]
2016-08-08 20:26:31,425 [       cli] [INFO] Highest CP Pokemon: Pidgeotto [CP: 765] [IV: 8/3/5] Potential: 0.36
2016-08-08 20:26:31,425 [       cli] [INFO] Most Perfect Pokemon: Magikarp [CP: 97] [IV: 12/15/13] Potential: 0.89
Traceback (most recent call last):
  File "pokecli.py", line 503, in <module>
    main()
  File "pokecli.py", line 89, in main
    bot.tick()
  File "/home/bjorn/dev/PokemonGo-Bot/pokemongo_bot/__init__.py", line 434, in tick
    if worker.work() == WorkerResult.RUNNING:
  File "/home/bjorn/dev/PokemonGo-Bot/pokemongo_bot/cell_workers/move_to_fort.py", line 82, in work
    if not step_walker.step():
  File "/home/bjorn/dev/PokemonGo-Bot/pokemongo_bot/step_walker.py", line 57, in step
    self.bot.heartbeat()
  File "/home/bjorn/dev/PokemonGo-Bot/pokemongo_bot/__init__.py", line 978, in heartbeat
    self.update_web_location()  # updates every tick
  File "/home/bjorn/dev/PokemonGo-Bot/pokemongo_bot/__init__.py", line 492, in update_web_location
    gym_longitude=fort.get('longitude')
  File "/home/bjorn/dev/PokemonGo-Bot/src/pgoapi/pgoapi/pgoapi.py", line 122, in function
    return request.call()
  File "/home/bjorn/dev/PokemonGo-Bot/pokemongo_bot/api_wrapper.py", line 106, in call
    result = self._call()
  File "/home/bjorn/dev/PokemonGo-Bot/pokemongo_bot/api_wrapper.py", line 63, in _call
    return PGoApiRequest.call(self)
  File "/home/bjorn/dev/PokemonGo-Bot/src/pgoapi/pgoapi/pgoapi.py", line 220, in call
    response = request.request(self._api_endpoint, self._req_method_list, self.get_position())
  File "/home/bjorn/dev/PokemonGo-Bot/src/pgoapi/pgoapi/rpc_api.py", line 129, in request
    response_dict = self._parse_main_response(response, subrequests)
  File "/home/bjorn/dev/PokemonGo-Bot/src/pgoapi/pgoapi/rpc_api.py", line 359, in _parse_main_response
    response_proto_dict = self._parse_sub_responses(response_proto, subrequests, response_proto_dict)
  File "/home/bjorn/dev/PokemonGo-Bot/src/pgoapi/pgoapi/rpc_api.py", line 376, in _parse_sub_responses
    request_entry = subrequests_list[i]
IndexError: list index out of range
```

**Comment by AllYourBasesAreBelongToUs** [comment]
No, I don't think it is a server issue. I think that they are actively messing with it, probably experimenting their next API modification, and making sure it weeds out the intended "customers".

**Comment by ocoot** [comment]
![screenshot from 2016-08-09 01-53-19](https://cloud.githubusercontent.com/assets/17609682/17491992/1aae3632-5dd4-11e6-9f4d-2f0ac5765784.png)

its working again now btw

**Comment by AllYourBasesAreBelongToUs** [comment]
Like I said, it seems to be account dependent (my bot C still crashes on this).
So, I fear that this is a prelude to the worse, and we might need the magic of 'wchill' (one of the core devs of last API fix for those who don't know) sooner than we might have liked...

**Comment by renatorib** [comment]
Same problem here

**Comment by AllYourBasesAreBelongToUs** [comment]
@ocoot I would recommend that your remove your image in attachement. It makes your account easily identifiable. Not that Niantic would care to address botting on a case-by-case basis, but you are better off remaining stealthy I think.

**Comment by ocoot** [comment]
ok bro. sorry :) @AllYourBasesAreBelongToUs

**Comment by demesg** [comment]
My "stable" bot does not have this issue, but my dev Bot has it ..

**Comment by AllYourBasesAreBelongToUs** [comment]
Yes, weird, it is on and off (but remains on for now on my bot C). But, maybe this is just a server issue after all, and I am just being paranoïd.

**Comment by JohnAlbert1** [comment]
Could be worth mentioning that out of my 4 bots the only one up and running is the one that is botting in my country. Perhaps location is being filtered by ip or something?

**Comment by renatorib** [comment]
5 minutes after my comment here (saying "same problem"), I just run "./run.sh" again and it working. Without updating anything.
It seems that this error appears occasionally. Weird.

**Comment by djadstar** [comment]
I'm seeing the same, first account running still. 'Bot' account getting the error.

**Comment by mrseeker** [comment]
Well, got myself a nice slew of messages, full report here:

```
Traceback (most recent call last):
  File "pokecli.py", line 503, in <module>
    main()
  File "pokecli.py", line 89, in main
    bot.tick()
  File "/home/pokemon/PokemonGo-Bot/pokemongo_bot/__init__.py", line 441, in tick
    if worker.work() == WorkerResult.RUNNING:
  File "/home/pokemon/PokemonGo-Bot/pokemongo_bot/cell_workers/spin_fort.py", line 44, in work
    player_longitude=f2i(self.bot.position[1])
  File "/home/pokemon/PokemonGo-Bot/src/pgoapi/pgoapi/pgoapi.py", line 122, in function
    return request.call()
  File "/home/pokemon/PokemonGo-Bot/pokemongo_bot/api_wrapper.py", line 106, in call
    result = self._call()
  File "/home/pokemon/PokemonGo-Bot/pokemongo_bot/api_wrapper.py", line 63, in _call
    return PGoApiRequest.call(self)
  File "/home/pokemon/PokemonGo-Bot/src/pgoapi/pgoapi/pgoapi.py", line 220, in call
    response = request.request(self._api_endpoint, self._req_method_list, self.get_position())
  File "/home/pokemon/PokemonGo-Bot/src/pgoapi/pgoapi/rpc_api.py", line 129, in request
    response_dict = self._parse_main_response(response, subrequests)
  File "/home/pokemon/PokemonGo-Bot/src/pgoapi/pgoapi/rpc_api.py", line 359, in _parse_main_response
    response_proto_dict = self._parse_sub_responses(response_proto, subrequests, response_proto_dict)
  File "/home/pokemon/PokemonGo-Bot/src/pgoapi/pgoapi/rpc_api.py", line 376, in _parse_sub_responses
    request_entry = subrequests_list[i]
IndexError: list index out of range
2016-08-08 15:19:33,763 [sentry.errors] [ERROR] Sentry responded with an error: 'module' object has no attribute 'HTTPSHandler' (url: https://app.getsentry.com/api/90254/store/)
Traceback (most recent call last):
  File "/home/pokemon/PokemonGo-Bot/local/lib/python2.7/site-packages/raven/transport/threaded.py", line 174, in send_sync
    super(ThreadedHTTPTransport, self).send(data, headers)
  File "/home/pokemon/PokemonGo-Bot/local/lib/python2.7/site-packages/raven/transport/http.py", line 47, in send
    ca_certs=self.ca_certs,
  File "/home/pokemon/PokemonGo-Bot/local/lib/python2.7/site-packages/raven/utils/http.py", line 44, in urlopen
    class ValidHTTPSHandler(urllib2.HTTPSHandler):
AttributeError: 'module' object has no attribute 'HTTPSHandler'
2016-08-08 15:19:33,764 [sentry.errors.uncaught] [ERROR] [u'IndexError: list index out of range', u'  File "pokecli.py", line 503, in <module>', u'  File "pokecli.py", line 89, in main', u'  File "pokemongo_bot/__init__.py", line 441, in tick', u'  File "pokemongo_bot/cell_workers/spin_fort.py", line 44, in work', u'  File "pgoapi/pgoapi.py", line 122, in function', u'  File "pokemongo_bot/api_wrapper.py", line 106, in call', u'  File "pokemongo_bot/api_wrapper.py", line 63, in _call', u'  File "pgoapi/pgoapi.py", line 220, in call', u'  File "pgoapi/rpc_api.py", line 129, in request', u'  File "pgoapi/rpc_api.py", line 359, in _parse_main_response', u'  File "pgoapi/rpc_api.py", line 376, in _parse_sub_responses']
```
Dev commit: f648be31c15d0143bccc0531571b2d00fd88ec4d

**Comment by MrKaralis** [comment]
@JohnAlbert1 I've the same problem and the bot is set to work 20km away from my current position

**Comment by mrseeker** [comment]
From the looks of it, it seems like they are messing with the API again to throw the bots out (since Unknown6 is already known).

**Comment by firzzen** [comment]
For me i've got three bots working fine for the last hour or so, but one just won't work at all. I tried the following:

1) delete last location data and cell data from /data folder
2) Remove this part from the config:

        "config":{
          "lure_attraction": true,
          "lure_max_distance": 2000
        }
3) Set health_record to false in the config.

The following worked for one bot but not for another. Here is the error:

2016-08-08 22:22:40,188 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Lake Pit Viewing Station (71c496edb8a945c884f45200ab1d821e.16)
2016-08-08 22:22:47,451 [       cli] [INFO]
2016-08-08 22:22:47,451 [       cli] [INFO] Ran for 0:06:34
2016-08-08 22:22:47,452 [       cli] [INFO] Total XP Earned: 780  Average: 7132.47/h
2016-08-08 22:22:47,452 [       cli] [INFO] Travelled 0.01km
2016-08-08 22:22:47,452 [       cli] [INFO] Visited 3 stops
2016-08-08 22:22:47,452 [       cli] [INFO] Encountered 3 pokemon, 3 caught, 0 released, 0 evolved, 0 never seen before
2016-08-08 22:22:47,453 [       cli] [INFO] Threw 5 pokeballs
2016-08-08 22:22:47,453 [       cli] [INFO] Earned 300 Stardust
2016-08-08 22:22:47,453 [       cli] [INFO]
2016-08-08 22:22:47,453 [       cli] [INFO] Highest CP Pokemon: Cubone [CP: 433] [IV: 14/4/11] Potential: 0.64
2016-08-08 22:22:47,454 [       cli] [INFO] Most Perfect Pokemon: Ponyta [CP: 403] [IV: 11/10/8] Potential: 0.64
Traceback (most recent call last):
  File "pokecli.py", line 501, in <module>
    main()
  File "pokecli.py", line 132, in main
    raise e
IndexError: list index out of range

**Comment by MrKaralis** [comment]
Now everything is working fine again a real strange behaviour.

**Comment by bjchristenson** [comment]
For what it's worth, I have scripts that run each bot in a term window and launch with an infinite loop so if a bot craps out, it automatically restarts.

**Comment by leolambertini** [comment]
I did this under "MoveToFort":

`lure_attraction": **false**`

Seems to do the trick for me, no more crashing for over 10 minutes.

**Comment by ocoot** [comment]
everything works now without tricking the code :dancer:

**Comment by whitmanc** [comment]
Working on my end now, was having same issues earlier. We'll see what happens.......

**Comment by firzzen** [comment]
The bot hasn't crashed but now what happens is this:

**Bot number 1:**

2016-08-08 22:57:15,894 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop S. M. U. R. R. F. Tree Tribute Mural - 0.09km (attraction of lure 0.12km)
2016-08-08 22:57:18,488 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Palisades Park (554204bdc6cb4733955f11f7e6d57b8a.16)
2016-08-08 22:57:24,169 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop S. M. U. R. R. F. Tree Tribute Mural - 0.08km (attraction of lure 0.12km)
2016-08-08 22:57:26,433 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Palisades Park (554204bdc6cb4733955f11f7e6d57b8a.16)
2016-08-08 22:57:31,769 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop S. M. U. R. R. F. Tree Tribute Mural - 0.06km (attraction of lure 0.11km)
2016-08-08 22:57:34,372 [CatchLuredPokemon] [INFO] [lured_pokemon_found] Lured pokemon at fort Palisades Park (554204bdc6cb4733955f11f7e6d57b8a.16)
2016-08-08 22:57:39,742 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop S. M. U. R. R. F. Tree Tribute Mural - 0.05km (attraction of lure 0.10km)
2016-08-08 22:57:42,844 [  SpinFort] [INFO] [pokestop_out_of_range] Pokestop S. M. U. R. R. F. Tree Tribute Mural out of range.
2016-08-08 22:57:42,844 [  SpinFort] [INFO] [pokestop_searching_too_often] Possibly searching too often, take a rest.
2016-08-08 22:57:42,856 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Santa Monica Pier Aquarium - 0.08km (attraction of lure 0.08km)
2016-08-08 22:57:44,719 [MoveToFort] [INFO] [moving_to_lured_fort] Moving towards pokestop Santa Monica Pier Aquarium - 0.07km (attraction of lure 0.07km)


**Bot number 2:**

2016-08-08 22:52:07,167 [FollowSpiral] [INFO] [position_update] Walking from (34.061419122967344, -118.35571762514489, 0) to (34.061419122967344, -118.35635875771466, 0) (59.0599173102 m)
2016-08-08 22:52:14,112 [FollowSpiral] [INFO] [position_update] Walking from (34.061419122967344, -118.35635875771466, 0) to (34.061419122967344, -118.35699989028443, 0) (59.0599173102 m)
2016-08-08 22:52:22,399 [FollowSpiral] [INFO] [position_update] Walking from (34.061419122967344, -118.35699989028443, 0) to (34.061419122967344, -118.3576410228542, 0) (59.0599173102 m)
2016-08-08 22:52:29,652 [FollowSpiral] [INFO] [position_update] Walking from (34.061419122967344, -118.3576410228542, 0) to (34.061419122967344, -118.35828215542396, 0) (59.0599173102 m)
2016-08-08 22:52:36,357 [FollowSpiral] [INFO] [position_update] Walking from (34.061419122967344, -118.35828215542396, 0) to (34.06205193822608, -118.35828215542396, 0) (70.3658225492 m)
2016-08-08 22:52:45,127 [FollowSpiral] [INFO] [position_update] Walking from (34.06205193822608, -118.35828215542396, 0) to (34.06268475348481, -118.35828215542396, 0) (70.3658225492 m)
2016-08-08 22:52:54,360 [FollowSpiral] [INFO] [position_update] Walking from (34.06268475348481, -118.35828215542396, 0) to (34.06331756874354, -118.35828215542396, 0) (70.3658225492 m)
2016-08-08 22:53:03,498 [FollowSpiral] [INFO] [position_update] Walking from (34.06331756874354, -118.35828215542396, 0) to (34.063950384002275, -118.35828215542396, 0) (70.3658225492 m)

And so forth, this happens with every bot now. It either gets stuck after a while with FollowSpiral or SpinFort.

**Comment by RaizenInstinct** [comment]
@firzzen do you have enough space in your pokemon bag?

**Comment by firzzen** [comment]
Yes of course @RaizenInstinct , it's multiple bots, not just one.

**Comment by firzzen** [comment]
Now it's working fine. Was walking, walking, walking, walking...being too far from pokestops etc but then suddenly all of them started working properly.

Must be a server related issue?

**Comment by ocoot** [comment]
bot crashed after trying to fix softban?


2016-08-09 07:17:17,414 [PokemonGoBot] [INFO] [bot_start] Starting bot...
2016-08-09 07:17:19,645 [CollectLevelUpReward] [INFO] [level_up_reward] Received level up reward: []
2016-08-09 07:17:19,646 [IncubateEggs] [INFO] [next_egg_incubates] Next egg incubates in 2.66 km
2016-08-09 07:17:22,629 [PokemonCatchWorker] [INFO] [pokemon_appeared] A wild Charmander appeared! [CP 423] [Potential 0.47] [A/D/S 1/8/12]
2016-08-09 07:17:26,285 [PokemonCatchWorker] [WARNING] [softban] Failed to use berry. You may be softbanned.
2016-08-09 07:17:28,355 [       cli] [INFO] 
2016-08-09 07:17:28,356 [       cli] [INFO] Ran for 0:01:11
2016-08-09 07:17:28,356 [       cli] [INFO] Total XP Earned: 0  Average: 0.00/h
2016-08-09 07:17:28,356 [       cli] [INFO] Travelled 0.00km
2016-08-09 07:17:28,356 [       cli] [INFO] Visited 0 stops
2016-08-09 07:17:28,357 [       cli] [INFO] Encountered 0 pokemon, 0 caught, 0 released, 0 evolved, 0 never seen before
2016-08-09 07:17:28,357 [       cli] [INFO] Threw 0 pokeballs
2016-08-09 07:17:28,357 [       cli] [INFO] Earned 0 Stardust
2016-08-09 07:17:28,357 [       cli] [INFO] 
2016-08-09 07:17:28,357 [       cli] [INFO] Highest CP Pokemon: 
2016-08-09 07:17:28,357 [       cli] [INFO] Most Perfect Pokemon: 
Traceback (most recent call last):
  File "pokecli.py", line 521, in <module>
    main()
  File "pokecli.py", line 95, in main
    bot.tick()
  File "/root/PokemonGo-Bot/pokemongo_bot/__init__.py", line 446, in tick
    if worker.work() == WorkerResult.RUNNING:
  File "/root/PokemonGo-Bot/pokemongo_bot/cell_workers/catch_visible_pokemon.py", line 36, in work
    return self.catch_pokemon(self.bot.cell['catchable_pokemons'].pop(0))
  File "/root/PokemonGo-Bot/pokemongo_bot/cell_workers/catch_visible_pokemon.py", line 48, in catch_pokemon
    return_value = worker.work()
  File "/root/PokemonGo-Bot/pokemongo_bot/cell_workers/pokemon_catch_worker.py", line 114, in work
    self._do_catch(pokemon, encounter_id, catch_rate_by_ball, is_vip=is_vip)
  File "/root/PokemonGo-Bot/pokemongo_bot/cell_workers/pokemon_catch_worker.py", line 332, in _do_catch
    'success_percentage': self._pct(catch_rate_by_ball[current_ball]),
IndexError: list index out of range

**Comment by bigkraig** [comment]
:+1:

**Comment by bigkraig** [comment]
there was a refactor of pokemon_catch_worker that may have introduced this

**Comment by bigkraig** [comment]
Try this

```
diff --git a/pokemongo_bot/cell_workers/pokemon_catch_worker.py b/pokemongo_bot/cell_workers/pokemon_catch_worker.py
index a0711b4..ae55b7d 100644
--- a/pokemongo_bot/cell_workers/pokemon_catch_worker.py
+++ b/pokemongo_bot/cell_workers/pokemon_catch_worker.py
@@ -252,6 +252,7 @@ class PokemonCatchWorker(BaseTask):
 
             # softban?
             else:
+                new_catch_rate_by_ball = catch_rate_by_ball
                 self.emit_event(
                     'softban',
                     level='warning',
@@ -260,6 +261,7 @@ class PokemonCatchWorker(BaseTask):
 
         # unknown status code
         else:
+            new_catch_rate_by_ball = catch_rate_by_ball
             self.emit_event(
                 'threw_berry_failed',
                 formatted='Unknown response when throwing berry: {status_code}.',
```

**Comment by viniciusbo** [comment]
Commenting https://github.com/PokemonGoF/PokemonGo-Bot/blob/master/pokecli.py#L84 seems to workaround this issue.

**Comment by bigkraig** [comment]
I just submitted #3197 to fix it

**Comment by J2501** [comment]
I use your fix but after a 2 minutes run, the issue came back but in an other file (perhaps the issue is different ?) :

`Traceback (most recent call last):
  File "pokecli.py", line 521, in <module>
    main()
  File "pokecli.py", line 81, in main
    bot.start()
  File "/home/sky/PokemonGo-Bot/pokemongo_bot/__init__.py", line 82, in start
    self._setup_api()
  File "/home/sky/PokemonGo-Bot/pokemongo_bot/__init__.py", line 690, in _setup_api
    self.update_web_location()
  File "/home/sky/PokemonGo-Bot/pokemongo_bot/__init__.py", line 504, in update_web_location
    gym_longitude=fort.get('longitude')
  File "/home/sky/PokemonGo-Bot/src/pgoapi/pgoapi/pgoapi.py", line 122, in function
    return request.call()
  File "/home/sky/PokemonGo-Bot/pokemongo_bot/api_wrapper.py", line 106, in call
    result = self._call()
  File "/home/sky/PokemonGo-Bot/pokemongo_bot/api_wrapper.py", line 63, in _call
    return PGoApiRequest.call(self)
  File "/home/sky/PokemonGo-Bot/src/pgoapi/pgoapi/pgoapi.py", line 220, in call
    response = request.request(self._api_endpoint, self._req_method_list, self.get_position())
  File "/home/sky/PokemonGo-Bot/src/pgoapi/pgoapi/rpc_api.py", line 129, in request
    response_dict = self._parse_main_response(response, subrequests)
  File "/home/sky/PokemonGo-Bot/src/pgoapi/pgoapi/rpc_api.py", line 359, in _parse_main_response
    response_proto_dict = self._parse_sub_responses(response_proto, subrequests, response_proto_dict)
  File "/home/sky/PokemonGo-Bot/src/pgoapi/pgoapi/rpc_api.py", line 376, in _parse_sub_responses
    request_entry = subrequests_list[i]
IndexError: list index out of range
`

**Comment by firzzen** [comment]
Can anyone comment on the fix from #3197 ? Has it improved? I probably should pull latest dev though, haven't done it since Sunday morning :)

**Comment by J2501** [comment]
The lastest dev with this fix runs like a charm since 2 hours.
( I probably fixed the file like a banana ... )
